webapp/yarn.lock

```diff
@@ -0,0 +1,7549 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```

Comment:
Why yarn over npm? and why not in the project root?

Reply:
That is a good question, I'd love to hear your thoughts on this.

The webapp and the server coexists in the same repo but are two different projects. They have their own different dependencies—and not much common ones.

Keeping all dependencies in the project root will quickly become cumbersome, that is why I separated `/webapp` package.json.

I do agree that keeping track of two different places to install dependencies is not ideal, and that is why I started using `yarn`.

Yarn has a feature called [workspaces](https://yarnpkg.com/lang/en/docs/workspaces/) that enables you to have a directory structure of multiple projects running their own dependencies while having only one place to run install: from the root.

To make use of yarn's workspace, we would have to adjust the project directories to look like this:
```
.
├── README.md
├── package.json
├── server
│   ├── package.json
│   ├── ...
├── webapp
│   ├── package.json
│   ├── ...
├── node_modules
```

We would then be able to run `yarn install` from the root, have all dependencies installed at the `/node_modules` at root and keeping server and client specific dependencies isolated from each other.

This is just a suggestion and I'm open to do in other ways.
I'm just not convinced that putting all dependencies at the root `package.json` as it is right now is the best way to go...

Reply:
> Keeping all dependencies in the project root will quickly become cumbersome.
Maybe for a large project that is the case, but this is a pretty simple project, with not too many deps, so its reasonable to keep them all together.

Reply:
Yes, I agree with you, like I mentioned below.